### PR TITLE
Averaging in `TallyRelativeError`

### PR DESCRIPTION
**Pull request by nuclearkevin**
Added averaging to `TallyRelativeError` and modified the fission_tally_relative_error tests to include averaging post-processors.
(Ref. #850)

### CODE REVIEW

File: src/postprocessors/TallyRelativeError.C [hunk 1]
```diff
@@ -86,20 +86,24 @@ TallyRelativeError::getValue() const
...
+      post_processor_value = std::numeric_limits<Real>::max();
+      break;
+    case operation::average:
+      post_processor_value = 0.0;
       break;
     default:
       mooseError("Unhandled OperationEnum!");
   }
 
+  unsigned int num_values = 0u;
```
Comment (aprilnovak) [inline src/postprocessors/TallyRelativeError.C:106]: I don't think you need this `0u`, but `0` would work fine. Is there a specific scenario you were thinking of by using `0u`, or just habit?

Comment (nuclearkevin) [reply src/postprocessors/TallyRelativeError.C:106]: It's a habit of mine from other projects with styles that abuse `auto`. I can change it to `0` and try to break the habit for future PRs if that's preferred.

Comment (aprilnovak) [reply src/postprocessors/TallyRelativeError.C:106]: I think convention in MOOSE is to use type-agnostic syntax as much as possible, so I'd prefer to remove it. I think you can also safely remove the `static_cast<Real>` line later as well, I think the type conversion should happen fine without it (but lmk if that's not the case).

Comment (nuclearkevin) [reply src/postprocessors/TallyRelativeError.C]: Pushed style changes. Tests are still passing on my end so it looks like the `static_cast` was unnecessary.

**Pull request merged**